src/les/index.ts

```diff
@@ -0,0 +1,245 @@
+    )
+    this._send(LES.MESSAGE_CODES.STATUS, rlp.encode(statusList))
+    this._handleStatus()
+  }
+
+  sendMessage(code: LES.MESSAGE_CODES, reqId: number, payload: any) {
+    debug(
+      `Send ${this.getMsgPrefix(code)} message to ${this._peer._socket.remoteAddress}:${
+        this._peer._socket.remotePort
+      }: ${rlp.encode(payload).toString('hex')}`,
+    )
+    switch (code) {
+      case LES.MESSAGE_CODES.STATUS:
+        throw new Error('Please send status message through .sendStatus')
+
+      case LES.MESSAGE_CODES.ANNOUNCE: // LES/1
+      case LES.MESSAGE_CODES.GET_BLOCK_HEADERS:
+      case LES.MESSAGE_CODES.BLOCK_HEADERS:
+      case LES.MESSAGE_CODES.GET_BLOCK_BODIES:
+      case LES.MESSAGE_CODES.BLOCK_BODIES:
+      case LES.MESSAGE_CODES.GET_RECEIPTS:
+      case LES.MESSAGE_CODES.RECEIPTS:
+      case LES.MESSAGE_CODES.GET_PROOFS:
+      case LES.MESSAGE_CODES.PROOFS:
+      case LES.MESSAGE_CODES.GET_CONTRACT_CODES:
+      case LES.MESSAGE_CODES.CONTRACT_CODES:
+      case LES.MESSAGE_CODES.GET_HEADER_PROOFS:
+      case LES.MESSAGE_CODES.HEADER_PROOFS:
+      case LES.MESSAGE_CODES.SEND_TX:
+      case LES.MESSAGE_CODES.GET_PROOFS_V2: // LES/2
+      case LES.MESSAGE_CODES.PROOFS_V2:
+      case LES.MESSAGE_CODES.GET_HELPER_TRIE_PROOFS:
+      case LES.MESSAGE_CODES.HELPER_TRIE_PROOFS:
+      case LES.MESSAGE_CODES.SEND_TX_V2:
+      case LES.MESSAGE_CODES.GET_TX_STATUS:
+      case LES.MESSAGE_CODES.TX_STATUS:
+        if (this._version >= LES.les2.version) break
+        throw new Error(`Code ${code} not allowed with version ${this._version}`)
+
+      default:
+        throw new Error(`Unknown code ${code}`)
+    }
+
+    this._send(code, rlp.encode([reqId, payload]))
+  }
+
+  getMsgPrefix(msgCode: LES.MESSAGE_CODES) {
+    return LES.MESSAGE_CODES[msgCode]
+  }
+}
+
+export namespace LES {
+  export interface Status {
+    [key: string]: Buffer | number
+    protocolVersion: Buffer
+    networkId: Buffer | number
+    headTd: Buffer
+    headHash: Buffer
+    headNum: Buffer
+    genesisHash: Buffer
+    serveHeaders: Buffer
+    serveChainSince: Buffer
+    serveStateSince: Buffer
+    txRelax: Buffer
+    'flowControl/BL': Buffer
+    'flowControl/MRR': Buffer
+    'flowControl/MRC': Buffer
+    announceType: Buffer | number
+  }
+
+  export enum MESSAGE_CODES {
+    // LES/1
+    STATUS = 0x00,
+    ANNOUNCE = 0x01,
+    GET_BLOCK_HEADERS = 0x02,
+    BLOCK_HEADERS = 0x03,
+    GET_BLOCK_BODIES = 0x04,
+    BLOCK_BODIES = 0x05,
+    GET_RECEIPTS = 0x06,
+    RECEIPTS = 0x07,
+    GET_PROOFS = 0x08,
+    PROOFS = 0x09,
+    GET_CONTRACT_CODES = 0x0a,
+    CONTRACT_CODES = 0x0b,
+    GET_HEADER_PROOFS = 0x0d,
+    HEADER_PROOFS = 0x0e,
+    SEND_TX = 0x0c,
+
+    // LES/2
+    GET_PROOFS_V2 = 0x0f,
+    PROOFS_V2 = 0x10,
+    GET_HELPER_TRIE_PROOFS = 0x11,
+    HELPER_TRIE_PROOFS = 0x12,
+    SEND_TX_V2 = 0x13,
+    GET_TX_STATUS = 0x14,
+    TX_STATUS = 0x15,
+  }
+}
```

Comment:
Same here, `MESSAGE_CODES` moved to its own `LES` namespace at the end of the file, looks good.